internal/app/machined/internal/phase/config/config.go

```diff
@@ -30,30 +37,101 @@ func (task *Task) TaskFunc(mode runtime.Mode) phase.TaskFunc {
 }
 
 func (task *Task) standard(r runtime.Runtime) (err error) {
+	_, err = os.Stat(constants.ConfigPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Printf("did not find existing config.yaml. pulling it")
+
+			var cfgBytes []byte
+
+			cfgBytes, err = fetchConfig(r)
+			if err != nil {
+				return err
+			}
+
+			*task.cfgBytes = cfgBytes
+
+			return nil
+		}
+
+		return err
+	}
+
+	log.Printf("found an existing config.yaml. determining if we should pull anyways")
+
+	forceCfgPull := false
+
+	content, err := config.FromFile(constants.ConfigPath)
+	if err != nil {
+		forceCfgPull = true
+
+		log.Printf("failed to read config, forcing a re-pull of the config: %s", err)
+	}
+
+	cfg, err := config.New(content)
```

Comment:
I think this should go under `else` for the `if err != nil` above, as `content` will be `nil` if we can't read the file

Reply:
yep, good spot. fixed in amended commit.